src/lazy/expanded/mod.rs

```diff
@@ -397,21 +497,23 @@ impl<Encoding: Decoder, Input: IonInput> ExpandingReader<Encoding, Input> {
     /// This method will consume and process as many system-level values as possible until it
     /// encounters an application-level value or the end of the stream.
     pub fn next_value(&mut self) -> IonResult<Option<LazyValue<Encoding>>> {
+        use SystemStreamItem::*;
         loop {
-            match self.next_item()? {
-                SystemStreamItem::VersionMarker(_marker) => {
-                    // TODO: Handle version changes 1.0 <-> 1.1
-                }
-                SystemStreamItem::SymbolTable(_) => {
-                    // The symbol table is processed by `next_item` before it is returned. There's
-                    // nothing to be done here.
-                }
-                SystemStreamItem::Value(value) => return Ok(Some(value)),
-                SystemStreamItem::EndOfStream(_) => return Ok(None),
-            }
+            match self.next_item() {
+                Ok(Value(value)) => return Ok(Some(value)),
+                Ok(EndOfStream(_)) => return Ok(None),
+                Ok(_) => {}
+                Err(e) => return Err(e),
+            };
```

Comment:
Per our offline discussion, this (and some of the weirdness that spurred the introduction of custom macros) came about because `?` was showing up in some profiler hotspots. I'd like to leave them in place for the sake of merging this PR, but am open to trying to re-introduce `?` in another PR while looking out for performance regressions.